### PR TITLE
feat(cursor): add CUR-007 to CUR-009 validation rules

### LINKED ISSUE #296: New Cursor rules: CUR-007 to CUR-009 (alwaysApply+globs, boolean type, description)
## Summary

Add 3 new validation rules for Cursor `.mdc` rule files.

### CUR-007: alwaysApply with redundant globs (MEDIUM)
When `alwaysApply: true` is set, the `globs` field is redundant -- the rule applies to every chat session regardless of file patterns. Warn and suggest removing `globs`.

### CUR-008: Invalid alwaysApply type (HIGH)
`alwaysApply` must be a boolean (`true`/`false`), not a string `"true"` or `"false"`. YAML parses quoted values as strings.

### CUR-009: Missing description for agent-requested rule (MEDIUM)
If a rule has no `alwaysApply` and no `globs`, it becomes an "agent-requested" rule where the agent decides relevance based on the `description`. If `description` is also missing, the agent has no way to know when to apply the rule.

## Implementation

- Extend `CursorValidator` in `crates/agnix-core/src/rules/cursor.rs`
- Add rules to `knowledge-base/rules.json` and `VALIDATION-RULES.md`
- Add test fixtures in `tests/fixtures/cursor/`

## References

- [Cursor Rules](https://cursor.com/docs/context/rules)
- `agent-knowledge/agent-config-optional-fields.md`

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary

Closes #296

Adds 3 new Cursor `.mdc` validation rules:

- **CUR-007** [MEDIUM]: `alwaysApply` with redundant `globs` - warns when `alwaysApply: true` is set alongside `globs` (which is ignored). Includes safe auto-fix to remove the redundant field.
- **CUR-008** [HIGH]: Invalid `alwaysApply` type - detects when `alwaysApply` is a quoted string (`"true"`/`"false"`) instead of a boolean. Uses a flexible `AlwaysApplyField` enum to parse both types without falling through to CUR-003 (generic YAML error).
- **CUR-009** [MEDIUM]: Missing description for agent-requested rule - when a rule has no `alwaysApply` and no `globs`, it becomes "agent-requested" and the agent uses the `description` to decide relevance. Warns when description is also missing.

## Changes

- `crates/agnix-core/src/rules/cursor.rs` - Extended `CursorValidator` with 3 new rules
- `crates/agnix-core/src/schemas/cursor.rs` - Added `AlwaysApplyField` enum for flexible bool/string parsing
- `locales/en.yml`, `es.yml`, `zh-CN.yml` - Added i18n messages for CUR-007 to CUR-009
- `knowledge-base/rules.json` + `crates/agnix-rules/rules.json` - Added 3 new rule entries
- `knowledge-base/VALIDATION-RULES.md` - Added rule documentation and updated counts
- `tests/fixtures/cursor-invalid/` - Added 3 fixture files
- `website/docs/rules/` - Added 3 generated rule docs, updated index
- Updated rule counts (124 -> 127) across all parity tests

## Test plan

- [x] 18 new unit tests covering positive, negative, and edge cases
- [x] All 1900+ existing tests pass
- [x] Parity tests pass (rules.json, SARIF, VALIDATION-RULES.md, fixtures, website docs)
- [x] Test fixture files for each new rule

### CODE REVIEW

File: crates/agnix-core/src/rules/cursor.rs [hunk 1]
```diff
@@ -216,6 +219,103 @@ impl Validator for CursorValidator {
...
+                        t!("rules.cur_007.message"),
+                    )
+                    .with_suggestion(t!("rules.cur_007.suggestion"));
+
+                    // Safe auto-fix: remove the globs line
+                    if let Some((start, end)) = line_byte_range(content, globs_line) {
+                        diagnostic = diagnostic.with_fix(Fix::delete(
+                            start,
+                            end,
+                            "Remove redundant globs field".to_string(),
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/rules/cursor.rs]: Fixed. The auto-fix now uses `yaml_block_byte_range()` which deletes the key line plus all indented continuation lines. Added a test `test_cur_007_autofix_deletes_array_globs_block` that verifies array-style globs are fully removed.

File: crates/agnix-core/src/rules/cursor.rs [hunk 2]
```diff
@@ -216,6 +219,103 @@ impl Validator for CursorValidator {
...
+            if config.is_rule_enabled("CUR-008") {
+                if let Some(ref always_apply) = schema.always_apply {
+                    if always_apply.is_string() {
+                        let always_apply_line = parsed
+                            .raw
+                            .lines()
+                            .enumerate()
+                            .find(|(_, line)| line.trim_start().starts_with("alwaysApply:"))
+                            .map(|(idx, _)| parsed.start_line + 1 + idx)
+                            .unwrap_or(parsed.start_line);
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/rules/cursor.rs]: Fixed. Extracted `find_field_line()` helper and replaced all 3 inline line-finding patterns with it (CUR-004, CUR-007, CUR-008).

File: crates/agnix-core/src/rules/cursor.rs [hunk 3]
```diff
@@ -216,6 +219,103 @@ impl Validator for CursorValidator {
...
+            // If no alwaysApply and no globs, the rule is "agent-requested" and
+            // the agent uses the description to decide when to apply it.
+            if config.is_rule_enabled("CUR-009") {
+                let has_always_apply = schema.always_apply.is_some();
+                let has_globs = schema.globs.is_some();
+                let has_description = schema
+                    .description
+                    .as_ref()
+                    .map(|d| !d.trim().is_empty())
+                    .unwrap_or(false);
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/rules/cursor.rs]: Fixed. Now uses `as_deref().unwrap_or("").trim().is_empty()` pattern.

File: crates/agnix-core/src/rules/cursor.rs [hunk 4]
```diff
@@ -216,6 +219,103 @@ impl Validator for CursorValidator {
...
+                    .with_suggestion(t!("rules.cur_007.suggestion"));
+
+                    // Safe auto-fix: remove the globs line
+                    if let Some((start, end)) = line_byte_range(content, globs_line) {
+                        diagnostic = diagnostic.with_fix(Fix::delete(
+                            start,
+                            end,
+                            "Remove redundant globs field".to_string(),
+                            true,
+                        ));
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/rules/cursor.rs]: Fixed with `yaml_block_byte_range()` which deletes the key line plus all subsequent indented lines. Test `test_cur_007_autofix_deletes_array_globs_block` verifies array-form globs are fully removed.
